src/main/webapp/app/modules/canvas/filter/filter-panel.tsx

```diff
@@ -6,26 +6,34 @@ import './filter-panel.scss';
 import Search from '@spectrum-icons/workflow/Search';
 import FilterElement from 'app/modules/canvas/filter/filter-element';
 import { Translate } from 'react-jhipster';
-import { applyFilter, clearFilter } from './filter.reducer';
-import PanelHeader from 'app/shared/components/panel-header';
-
-export interface IFilterPanelProp extends StateProps, DispatchProps {}
+import { applyFilter, applyFilterForShareLink, clearFilter, clearFilterForShareLink } from './filter.reducer';import PanelHeader from 'app/shared/components/panel-header';
 
+export interface IFilterPanelProp extends StateProps, DispatchProps, RouteComponentProps { }
+import PanelHeader from 'app/shared/components/panel-header';
+import { IFeature } from 'app/shared/model/feature.model';
+import { getViewFeaturesEntities } from 'app/entities/feature/feature.reducer';
+import { RouteComponentProps } from 'react-router-dom';
+ 
+export interface IFilterPanelProp extends StateProps, DispatchProps, RouteComponentProps { }
+ 
 const FilterPanel = (props: IFilterPanelProp) => {
-
-  // TODO : need to refector this code
-  
+ 
+ // TODO : need to refector this code
   const [isFilterMinimize, setFilterMinimize] = useState(true);
   const [isFilterPanelClose, setFilterPanelClose] = useState(props.isFilterOpen);
-
-  const removeFilter = () => {
-    props.clearFilter({}, props.visualmetadata, props.view);
-  };
+  const params = new URLSearchParams(props.location.search);
+  const visualizationId = params.get('visualizationId');
 
   useEffect(() => {
     setFilterPanelClose(props.isFilterOpen);
   }, [props.isFilterOpen]);
 
+  useEffect(() => {
+    if (props.view.id) {
+      props.getViewFeaturesEntities(props.view.id);
+    }
+  }, [props.view]);
+
```

Comment:
why you need useEffect..can you do it without useEffect

Reply:
share link page directly call without canvas page so need to get list for dimensions for filter that's why we need this once we get view info we fetch Features by view id